pkg/apis/uiplugin/v1alpha1/types.go

```diff
@@ -31,20 +31,79 @@ type UIPluginList struct {
 	Items           []UIPlugin `json:"items"`
 }
 
-// +kubebuilder:validation:Enum=Dashboards
+// +kubebuilder:validation:Enum=Dashboards;TroubleshootingPanel;DistributedTracing
 type UIPluginType string
 
 const (
 	// TypeDashboards deploys the Dashboards Dynamic Plugin for OpenShift Console.
 	TypeDashboards UIPluginType = "Dashboards"
+	// DistributedTracing deploys the Distributed Tracing Dynamic Plugin for the OpenShift Console
+	TypeDistributedTracing UIPluginType = "DistributedTracing"
+	// TroubleshootingPanel deploys the Troubleshooting Panel Dynamic Plugin for the OpenShift Console
+	TypeTroubleshootingPanel UIPluginType = "TroubleshootingPanel"
 )
 
+// TroubleshootingPanelConfig contains options for configuring the Troubleshooting Panel  plugin
+type TroubleshootingPanelConfig struct {
+	// Timeout is the maximum duration before a query timeout.
+	//
+	// The value is expected to be a sequence of digits followed by a unit suffix, which can be 's' (seconds)
+	// or 'm' (minutes).
+	//
+	// +optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="OCP Console Query Timeout",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:ocpConsoleTimeout"}
+	// +kubebuilder:validation:Pattern:="^([0-9]+)([sm]{1})$"
+	Timeout string `json:"timeout,omitempty"`
+	// korrel8r defines the Korrel8r instance that the troubleshooting panel plugin will connect to
+	//
+	// +optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Korrel8r Instance"
+	Korrel8r TroubleshootingPanelKorrel8rConfig `json:"korrel8r,omitempty"`
+}
+
+type TroubleshootingPanelKorrel8rConfig struct {
+	// Name of the korrel8r instance
+	//
+	// +optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Korrel8r Instance Name"
+	Name string `json:"name,omitempty"`
```

Comment:
Should it really be optional?

Reading the code we assume a default value of "korrel8r" for both but I'd rather make these fields mandatory to be explicit.

Reply:
The korrel8r deployment is being attached to the troubleshooting panel UIPlugin in #497 so this specific comment might be helpful to address in that PR.

Reply:
"Korrel8r" will always be deployed  when the Troubleshooting Panel is installed in the same namespace where the observability-operator is installed. So there is no need for any configuration.

Reply:
Removed it in https://github.com/rhobs/observability-operator/pull/497